### PR TITLE
[ISSUE-856] report non default namespace drive event

### LINKED ISSUE #856: When csi-baremetal is deployed on non-default namespace, the CSI baremetal application generates events in the default namespace
**Describe the bug**
When csi-baremetal is deployed in non-default namespace and when the disk fails, the csi-baremetal application generates events in the default namespace.

**Environment (please complete the following information):**
 - Hardware: Any
 - OS: Any
 - Version: 1.1.0

**To Reproduce**
1. Deploy csi-baremetal in deployed in non-default namespace
2. Disk failure
3. kubectl get events -A to check the DriveHealthFailure  event on default namespace.

**Expected behavior**
When csi-baremetal is deployed in non-default namespace and when the disk fails, the csi-baremetal application generates events in the non-default namespace.

**Screenshots**
[root@centos76 ~]# kubectl get events -A | grep 40S0A0KRF1QF
default 6s Error DriveHealthFailure drive/09275394-45ed-426f-a735-0aaaf67ab326 Drive health is: BAD, previous state: GOOD. Drive Details: SN='40S0A0KRF1QF', Model='ATA TOSHIBA MG06ACA8', Type='HDD', Size='8001000000000', Firmware='GA08', Slot='21', NodeName='master0.ocp4.carnelian.com'
[root@centos76 ~]#
**Additional context**

### PR DESCRIPTION
**Pull request by yanxiucai**
### Resolves #856 

_Describe your changes_
Update the code in https://github.com/dell/csi-baremetal/blob/master/pkg/events/recorder/recorder.go function makeEvent() to generate events for drives in the same namespace where CSI is installed.

## PR checklist
- [x] Add link to the issue  
- [x] Choose Project
- [x] Choose PR label
- [x] New unit tests added
- [x] Modified code has meaningful comments
- [ ] All TODOs are linked with the issues
- [ ] All comments are resolved

## Testing

1. When CSI is installed in non-default namespace, drive event reports on corresponding non-default namespace.
```
admin@logan-dirt:~> kubectl get pods -A | grep csi
atlantic          csi-baremetal-controller-6bf87d6b48-7v998               3/4     Running     0          16s
atlantic          csi-baremetal-node-2hz59                                3/4     Running     0          16s
atlantic          csi-baremetal-node-6pk8q                                3/4     Running     0          16s
atlantic          csi-baremetal-node-controller-79d68f5878-bq2tr          1/1     Running     0          16s
atlantic          csi-baremetal-node-t6r94                                3/4     Running     0          16s
atlantic          csi-baremetal-operator-68f59d584-dmrmw                  1/1     Running     0          36s
atlantic          csi-baremetal-se-patcher-9glwp                          1/1     Running     0          16s
atlantic          csi-baremetal-se-vp5cq                                  0/1     Running     0          16s
admin@logan-dirt:~>
admin@logan-dirt:~>  kubectl annotate drive 5597f431-cced-4383-bdf1-64c58938a123 health=bad  --overwrite
drive.csi-baremetal.dell.com/5597f431-cced-4383-bdf1-64c58938a123 annotated
admin@logan-dirt:~> kubectl get events -A | grep 5597f431-cced-4383-bdf1-64c58938a123
atlantic          16s         Warning   DriveHealthOverridden   drive/5597f431-cced-4383-bdf1-64c58938a123                  Drive health is overridden with: BAD, real state: GOOD. Drive Details: SN='8010A09XTAJR', Model='0x1179 Dell Express Flash CD5 7.68T SFF', Type='NVME', Size='7681000000000', Firmware='1.1.1', Slot='Slot 7 Bay 1', NodeName='murray-dirt'
atlantic          16s         Error     DriveHealthFailure      drive/5597f431-cced-4383-bdf1-64c58938a123                  Drive health is: BAD, previous state: GOOD. Drive Details: SN='8010A09XTAJR', Model='0x1179 Dell Express Flash CD5 7.68T SFF', Type='NVME', Size='7681000000000', Firmware='1.1.1', Slot='Slot 7 Bay 1', NodeName='murray-dirt'
```
2.  When CSI is installed in default namespace, drive event reports on default namespace
```
admin@logan-dirt:~> kubectl get pods -A | grep csi
default           csi-baremetal-controller-6bf87d6b48-mc9qg               4/4     Running     0          3h41m
default           csi-baremetal-node-controller-79d68f5878-sl2hq          1/1     Running     0          3h41m
default           csi-baremetal-node-dxcl6                                4/4     Running     0          3h41m
default           csi-baremetal-node-rbs4d                                4/4     Running     0          3h41m
default           csi-baremetal-node-xwcnd                                4/4     Running     0          3h41m
default           csi-baremetal-operator-68f59d584-kc4kc                  1/1     Running     0          3h42m
default           csi-baremetal-se-6pxpd                                  1/1     Running     0          3h41m
default           csi-baremetal-se-patcher-vhb25                          1/1     Running     0          3h41m
admin@logan-dirt:~>  kubectl annotate drive 5597f431-cced-4383-bdf1-64c58938a123 health=bad  --overwrite
drive.csi-baremetal.dell.com/5597f431-cced-4383-bdf1-64c58938a123 annotated
admin@logan-dirt:~> kubectl get events -A | grep 5597f431-cced-4383-bdf1-64c58938a123
default           10s         Warning   DriveHealthOverridden   drive/5597f431-cced-4383-bdf1-64c58938a123                  Drive health is overridden with: BAD, real state: GOOD. Drive Details: SN='8010A09XTAJR', Model='0x1179 Dell Express Flash CD5 7.68T SFF', Type='NVME', Size='7681000000000', Firmware='1.1.1', Slot='Slot 7 Bay 1', NodeName='murray-dirt'
default           10s         Error     DriveHealthFailure      drive/5597f431-cced-4383-bdf1-64c58938a123                  Drive health is: BAD, previous state: GOOD. Drive Details: SN='8010A09XTAJR', Model='0x1179 Dell Express Flash CD5 7.68T SFF', Type='NVME', Size='7681000000000', Firmware='1.1.1', Slot='Slot 7 Bay 1', NodeName='murray-dirt'
admin@logan-dirt:~>
```

### CODE REVIEW

**Comment by yanxiucai** [comment]
Looks like there some format issue on test part, upload the test log here,  
[test.txt](https://github.com/dell/csi-baremetal/files/8843717/test.txt)

File: pkg/events/recorder/recorder.go [hunk 1]
```diff
@@ -67,7 +68,12 @@ func (sr *SimpleRecorder) makeEvent(ref *v1.ObjectReference, labels map[string]s
 	t := metav1.NewTime(now)
 	namespace := ref.Namespace
 	if namespace == "" {
-		namespace = metav1.NamespaceDefault
```
Comment (mishoyama) [inline pkg/events/recorder/recorder.go:70]: Do we really need to have this check on every `makeEvent`? Can we init global var with namespace on startup and use it?

**Comment by mishoyama** [comment]
@yanxiucai please fix DCO check

**Comment by yanxiucai** [comment]
Close this RP due to fmt and DCO failed. Create a new branch  for code change and passed DCO.  Refer new PR https://github.com/dell/csi-baremetal/pull/875

**Pull request closed without merging**